src/panels/calendar/ha-panel-calendar.ts

```diff
@@ -0,0 +1,304 @@
+class PanelCalendar extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property({ type: Boolean, reflect: true, attribute: "narrow" })
+  public narrow = false;
+
+  @property() private _calendars: SelectedCalendar[] = [];
+
+  @property() private _events: CalendarEvent[] = [];
+
+  private _start?: Date;
+
+  private _end?: Date;
+
+  private _firstUpdate = false;
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+    if (!this.hass) {
+      return;
+    }
+
+    this._fetchCalendars();
+  }
+
+  protected updated() {
+    if (!this._firstUpdate) {
+      this._fetchData();
+    }
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot="header"
+          ><app-toolbar>
+            <ha-menu-button
+              .hass=${this.hass}
+              .narrow=${this.narrow}
+            ></ha-menu-button>
+            <div main-title>${this.hass.localize("panel.calendar")}</div>
+          </app-toolbar>
+        </app-header>
+        <div class="content">
+          <div class="calendar-list">
+            <div class="calendar-list-header">My Calendars</div>
+            ${this._calendars.map(
+              (selCal) =>
+                html`<div>
+                  <mwc-formfield .label=${selCal.calendar.name}>
+                    <mwc-checkbox
+                      style=${styleMap({
+                        "--mdc-theme-secondary": selCal.backgroundColor,
+                      })}
+                      .value=${selCal.calendar.entity_id}
+                      .checked=${selCal.selected}
+                      @change=${this._handleToggle}
+                    ></mwc-checkbox>
+                  </mwc-formfield>
+                </div>`
+            )}
+          </div>
+          <ha-full-calendar
+            .events=${this._events}
+            @view-changed=${this._handleViewChanged}
+          ></ha-full-calendar>
+        </div>
+      </app-header-layout>
+    `;
+  }
+
+  private _fetchCalendars() {
+    this.hass.callApi<Calendar[]>("GET", "calendars").then((result) => {
+      this._calendars = result.map((cal, idx) => ({
+        selected: true,
+        calendar: cal,
+        backgroundColor: `#${palette[idx]}`,
+      }));
+    });
+  }
+
+  private _fetchData() {
+    if (!this._start || !this._end) {
+      return;
+    }
+
+    if (!this._firstUpdate && this._calendars.length) {
+      this._firstUpdate = true;
+    }
+
+    const start = new Date(this._start);
+    const end = new Date(this._end);
+    const params = encodeURI(
+      `?start=${start.toISOString()}&end=${end.toISOString()}`
+    );
+
+    const calEvents: CalendarEvent[] = [];
+    const calls = new Promise((resolve) => {
+      this._calendars
+        .filter((selCal) => selCal.selected)
+        .forEach(async (selCal, idx) => {
+          const events = await this.hass.callApi<any[]>(
```

Comment:
You are now getting the calendars one by one, use `await Promise.all()` to fetch them in parallel. Also means you don't need to do the forEach async and make it a promise.